### PR TITLE
PEP 621: Migrate from setup.py and setup.cfg to pyproject.toml

### PR DESCRIPTION
**Pull request by cclauss**
Migrate `setup.py` to `setup.cfg` using [setuptools-py2cfg](https://pypi.org/project/setuptools-py2cfg) plus manual modifications.  Then migrate `setup.cfg` to `pyproject.toml` using [ini2toml](https://pypi.org/project/ini2toml) to do the file conversion and running [pyproject-fmt](https://pypi.org/project/pyproject-fmt) and then [validate-pyproject](https://github.com/abravalheri/validate-pyproject) in pre-commit to validate the results.

### CODE REVIEW

**Comment by claudep** [comment]
Ideally, it would be nice to first fix the test suite before continuing modernizing linting/building. What do you think?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -63,7 +63,8 @@ jobs:
           python -m pip install --upgrade pip setuptools wheel
           pip install coverage requests tox tox-gh-actions
           pip install django==${{ matrix.django-version }} elasticsearch==${{ matrix.elastic-version }}
-          python setup.py clean build install
+          # python setup.py clean build install
```
Comment (claudep) [inline .github/workflows/test.yml:66]: Do you intend to keep this commented line?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -20,8 +20,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix: # https://docs.djangoproject.com/en/stable/faq/install/#what-python-version-can-i-use-with-django
-        django-version: ["3.2", "4.2", "5.0"]
-        python-version: ["3.8", "3.9"] # , "3.10", "3.11", "3.12"]  # Whoosh issues with Py3.10+
+        django-version: ["3.2", "4.2"] # , "5.0"]
```
Comment (claudep) [inline .github/workflows/test.yml:23]: For what reason did you have to remove "5.0" from the matrix?

**Review by claudep — COMMENTED** [review]
Looks rather good, thanks!

File: .github/workflows/test.yml (hunk 2 above)
Comment (cclauss) [reply .github/workflows/test.yml:23]: It currently fails.

File: .github/workflows/test.yml (hunk 1 above)
Comment (cclauss) [inline .github/workflows/test.yml:66]: Removed
```suggestion
```

**Review by claudep — APPROVED** [review]

**Pull request merged**